### PR TITLE
Fix rebasing over merge commits

### LINKED ISSUE #2160: Amending merge commit results in merge conflicts
**Describe the bug**
Amending merge commit results in merge conflicts.

**To Reproduce**
Try to amend a merge commit by making changes then selecting merge commit in the commits panel and using the keyboard shortcut `A`

**Expected behavior**
The amend is successful without merge conflicts, seeing as the merge conflicts were already handled when the original merge commit was made.

**Screenshots**
Example using this repo where the latest commit is a merge commit: https://github.com/OliverJAsh/lazygit-amend-merge-test

https://user-images.githubusercontent.com/921609/189199590-264cc0d7-2dd9-4bff-bfa8-63dd6db83103.mov

**Version info:**
```sh-session
$ lazygit --version
commit=, build date=, build source=homebrew, version=0.35, os=darwin, arch=arm64

$ git --version
git version 2.37.3
```

### PR DESCRIPTION
**Pull request by mark2185**
- **PR Description**

This should fix #2160 .

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

File: pkg/integration/components/input.go [hunk 1]
```diff
@@ -76,6 +76,12 @@ func (self *Input) Confirm() {
 	self.pressKey(self.keys.Universal.Confirm)
 }
 
+func (self *Input) ProceedWhenAsked(matcher *matcher) {
```
Comment (jesseduffield) [inline pkg/integration/components/input.go:79]: adding this because we do this all over the place

**Review by jesseduffield — APPROVED** [review]
LGMT: lemme know if you have thoughts on any of the tweaks I pushed

**Comment by jesseduffield** [comment]
hmm does that test also fail locally?

File: pkg/integration/components/input.go (hunk 1 above)
Comment (mark2185) [reply pkg/integration/components/input.go:79]: Smart

**Comment by mark2185** [comment]
> hmm does that test also fail locally?

Not for me. Let's see if a rebase fixes it.

File: pkg/integration/components/input.go (hunk 1 above)
Comment (jesseduffield) [reply pkg/integration/components/input.go:79]: https://www.youtube.com/watch?v=Ll9nfNlDLII&ab_channel=RapPadB

**Comment by jesseduffield** [comment]
Nice work

**Pull request merged**